### PR TITLE
Add info to sections ML-RSEs and other important tasks

### PR DESCRIPTION
**Pull request by ljgarcia**

### CODE REVIEW

File: competencies.md [hunk 1]
```diff
@@ -598,7 +598,8 @@ Beyond that, we feel that today other important tasks of RSEs are
...
   The RSE needs to be able to identify the application class they are dealing with and apply the respective RSE practices.
-- Enforcing reproducibility. Projects like ReproHack [@ReproHack] can greatly help in fostering that competency.
+- Enforcing reproducibility. Projects like ReproHack [@ReproHack] can greatly help in fostering that competency. 
+- Understanding metadata for research and, in particular, research software. There are ongoing efforts on metadata for research software such as CodeMeta [@jones_codemeta_2016], [Bioschemas Computational Tool](https://bioschemas.org/profiles/ComputationalTool/) and the \ac{NFDI} working group [@castro_research_sw_metadata_2023] on the subject. Other efforts focus on Software Management Plans (e.g., [@alves_elixir_2021], [@martinez_ortiz_practical_2022]) which could be helpful for RSEs at early stages (i.e., with no much experience on project management) as they give quick hints on what to look for regarding to basic management for research software (including information on e.g., licenses, releases, publication, citation) together with some ongoing work on corresponding metadata [@giraldo_metadata_2023].
```
Comment (CaptainSifff) [inline competencies.md:602]: I admit, that the metadata part is kind of a blind spot of the current competencies paper. For your contribution, I wonder whether we can shorten it slightly to have less references? A lot of them are definitely useful in the context of the survey project.

Comment (CaptainSifff) [reply competencies.md:602]: and the fixes to our use of ML are certainly warranted.

File: competencies.md [hunk 2]
```diff
@@ -1049,7 +1050,7 @@ New areas of expertise
 This RSE is interested in \glspl{SysOp} and system administration and sets up \ac{IT} infrastructures for and with researchers.
 Therefore, this specialisation on the one hand requires a deep knowledge of physical computer and network hardware and
 on the other hand knowledge about setup and configuration of particular server software, 
-e.g., setup of virtual machines on hypervisors or the planning and setup of compute server clusters machine learning.
+e.g., setup of virtual machines on hypervisors or the planning and setup of compute server clusters \ac{ML}.
```
Comment (jcohen02) [inline competencies.md:1053]: Should this say "for ML"?

```suggestion
e.g., setup of virtual machines on hypervisors or the planning and setup of compute server clusters for \ac{ML}.
```

**Review by jcohen02 — APPROVED** [review]
Great additions. I spotted a couple of typos and made one suggested minor update.

**Comment by CaptainSifff** [comment]
@mhagdorn , @jcohen02 would you mind working a bit on the refs?

File: competencies.md [hunk 3]
```diff
@@ -598,7 +598,8 @@ Beyond that, we feel that today other important tasks of RSEs are
...
   The RSE needs to be able to identify the application class they are dealing with and apply the respective RSE practices.
-- Enforcing reproducibility. Projects like ReproHack [@ReproHack] can greatly help in fostering that competency.
+- Enforcing reproducibility. Projects like ReproHack [@ReproHack] can greatly help in fostering that competency. 
+- Understanding metadata for research and, in particular, research software. There are ongoing efforts on metadata for research software such as CodeMeta [@jones_codemeta_2016], [Bioschemas Computational Tool](https://bioschemas.org/profiles/ComputationalTool/) and the \ac{NFDI} working group [@castro_research_sw_metadata_2023] on the subject. Other efforts focus on Software Management Plans (e.g., [@alves_elixir_2021], [@martinez_ortiz_practical_2022]) which could be helpful for RSEs at early stages (i.e., with not much experience of project management). They give quick hints on what to look for regarding basic management for research software (including information on e.g., licenses, releases, publication, citation) together with some ongoing work on corresponding metadata [@giraldo_metadata_2023].
```
[the remark this answers is not in the document]
Comment (jngrad) [reply competencies.md]: Not entirely sure if I'm leaving the scope of the competencies paper, but there is also the process of *FAIRification* of research data, that requires research software maintainers to become aware of ontologies available in their field and to update their software to comply with said ontologies when writing data to disk. For a recent example in materials science: Ghiringhelli et al. 2023 [doi:10.1038/s41597-023-02501-8](https://doi.org/10.1038/s41597-023-02501-8) and [CECAM workshop 1228](https://www.cecam.org/workshop-details/1228).

**Comment by sdruskat** [comment]
Great to see this PR, I had the same inkling about metadata missing, and will try to do some edits around this in the next days.
I'll work against the PR branch unless someone tells me not to?

**Comment by mhagdorn** [comment]
> Great to see this PR, I had the same inkling about metadata missing, and will try to do some edits around this in the next days. I'll work against the PR branch unless someone tells me not to?

excellent, thanks. go ahead we can figure out any conflicts should they arise

File: competencies.md [hunk 4]
```diff
@@ -1090,11 +1099,13 @@ compiled software.-->
...
-ML-RSEs analyse and check the suitability of an algorithm if it fulfils the needs of a certain task and they play a main role in deciding and selecting machine learning libraries for a given task.
-The increasing usage of ML in numerous scientific areas with social impact involves an emphasised awareness and consideration of possible manipulative or discriminatory influences.
+This involves knowledge about data analysis and feature engineering, metrics that are involved in \ac{ML}, \ac{ML} algorithm selection and cross validation, and knowledge in mathematical optimisation methods and statistics.
+Here, we use \ac{ML} in a broad sense of machine-based learning including deep learning, reinforcement learning, neuro-symbolic learning and similar.
+
+ML-RSEs analyse and check the suitability of an algorithm if it fulfils the needs of a certain task and they play a main role in deciding and selecting \ac{ML} libraries for a given task.
+The increasing usage of \ac{ML} in numerous scientific areas with social impact involves an emphasised awareness and consideration of possible manipulative or discriminatory influences.
```
Comment (jpthiele) [inline competencies.md:1106]: Maybe 'influences and biases', as bias is the correct term for many of the caveats of ML.

**Review by jpthiele — APPROVED** [review]
Could do with a few more line breaks to make editing easier
and less conflicting in the future.
Apart from that it's good.
Thank you for the additions!

File: competencies.md [hunk 5]
```diff
@@ -598,7 +598,16 @@ Beyond that, we feel that today other important tasks of RSEs are
...
+- Enforcing reproducibility. Projects like ReproHack [@ReproHack] can greatly help in fostering that competency. 
+- Understanding metadata for research and, in particular, research software.
+  There are ongoing efforts on metadata for research software such as CodeMeta [@jones_codemeta_2016],
+  [Bioschemas Computational Tool](https://bioschemas.org/profiles/ComputationalTool/)
+  and the \ac{NFDI} working group [@castro_research_sw_metadata_2023] on the subject.
+  Other efforts focus on Software Management Plans (e.g., [@alves_elixir_2021; @martinez_ortiz_practical_2022])
+  which could be helpful for RSEs at early stages (i.e., with not much experience of project management).
+  They give quick hints on what to look for regarding basic management for research
+  software (including information on e.g., licenses, releases, publication, citation)
+  together with some ongoing work on corresponding metadata [@giraldo_metadata_2023].
```
Comment (CaptainSifff) [inline competencies.md:610]: I'd like to add, that at some point this paragraph becomes a pointer to the literature, and moves away from the task of this section to exemplify todays task. I wonder why some of the documents couldn't be moved to Bioinformatics Skills section? I'm pretty sure, we could pick up the Metadata thread over there, too.
A course on Metadata in Kay's story could also help.

File: competencies.md (hunk 3 above)
[the remark this answers is not in the document]
Comment (jcohen02) [reply competencies.md]: This is a good point @jngrad - my view is that from a competencies perspective practitioners need to understand FAIR in the context of both data and software. It's important that they have a sufficient understanding to be able to actively and effectively contribute to development tasks in this area but there's, perhaps, a question about how far from the core competencies aspect we want to go before we start to lose focus a bit. No problem from my side with adding the suggested refs in but I think we should ensure that are supported by clear and defined context so it doesn't feel like we're opening a new space for discussion but then not actually providing sufficient insight. As I say, this is my perspective but I'm happy to go with the consensus within the group on how best to address this.

**Comment by sdruskat** [comment]
> excellent, thanks. go ahead we can figure out any conflicts should they arise

☝️ #221.

File: competencies.md (hunk 5 above)
Comment (sdruskat) [reply competencies.md:610]: > A course on Metadata in Kay's story could also help.

I've added some metadata stuff to her journey in https://github.com/CaptainSifff/paper_teaching-learning-RSE/pull/221/commits/a6910d34a1cdc62bf09ca4d2197a57057ef0436d as part of #221, although it isn't a full course. 

And to be honest, a course on software metadata alone I don't see in an RSE curriculum. Perhaps rather as part of a FAIR/SMP course.

File: competencies.md (hunk 1 above)
Comment (MakisH) [reply competencies.md]: The updated length and number of references seems appropriate to me:

> - Understanding metadata for research and research software.
  There are ongoing efforts on metadata for research software such as CodeMeta [@jones_codemeta_2016]
  and the \ac{NFDI} working group [@castro_research_sw_metadata_2023] on the subject.
  Metadata can also be used actively during and within a research project to inform the decision making processes [@Bird2016].

File: competencies.md (hunk 4 above)
Comment (mhagdorn) [reply competencies.md]: updated document as suggested

**Pull request merged**